sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JsonInferSchema.scala

```diff
@@ -99,6 +102,13 @@ private[sql] class JsonInferSchema(options: JSONOptions) extends Serializable {
             val wrappedCharException = new CharConversionException(msg)
             wrappedCharException.initCause(e)
             handleJsonErrorsByParseMode(parseMode, columnNameOfCorruptRecord, wrappedCharException)
+          case e: FileNotFoundException if ignoreMissingFiles =>
+            logWarning("Skipped missing file", e)
+            Some(StructType(Nil))
+          case e: FileNotFoundException if ignoreMissingFiles => throw e
+          case e @ (_: IOException | _: RuntimeException) if ignoreCorruptFiles =>
```

Comment:
refer https://github.com/apache/spark/pull/42979#discussion_r1360625931

Reply:
I understand that is an advice about code format. I am interested in what `RuntimeException` will happen.

Reply:
It is not for code format. It is an advice for follow catch rules of `FileScanRDD`. The purpose is to align with `FileScanRDD`'s processing logic on corrupt files (PS: To be honest, I'm not sure when the RuntimeException will occur, just to align the catch range of `FileScanRDD`)

Reply:
Yeah. That advice just for follow the catch rules(Personally, it's related to code format).
Because the origin code catches `RuntimeException`, ignore the comment above is also acceptable.